### PR TITLE
Feature/617 daterange picker and advances trainings

### PR DESCRIPTION
**Pull request by kcinay055679**

### CODE REVIEW

File: config/locales/rails.de.yml [hunk 1]
```diff
@@ -0,0 +1,224 @@
+de:
```
Comment (Robin481) [inline config/locales/rails.de.yml:1]: Simplify pl0x

File: spec/controllers/advanced_trainings_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      expect(response).to redirect_to(person_path(person))
+    end
+
+    it 'Create action creates a new advanced training for person and renders parent person' do
+      description = 'This is a description'
+      month_from = 1
+      year_from = 2020
+      month_to = 1
+      year_to = 2023
+      post :create , params: {:person_id=> person, advanced_training: { month_from: month_from, year_from: year_from, month_to: month_to, year_to: year_to, description: description} }
```
Comment (Robin481) [inline spec/controllers/advanced_trainings_controller_spec.rb:29]: Do this on multiple lines!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

File: spec/features/core_competences_spec.rb [hunk 3]
```diff
@@ -42,6 +42,7 @@
   it 'should display skill with same parent category in same row with divider' do
     visit person_path(people(:alice))
     expect(page).to have_selector('.circle-divider')
-    expect(page).to have_selector('.core-competence', count: 1, text: "Software-Engineering\nRails\nember")
+    require 'pry'; binding.pry # rubocop:disable Style/Semicolon,Lint/Debugger
```
Comment (Robin481) [inline spec/features/core_competences_spec.rb:45]: No prying eyes

File: spec/models/advanced_training_spec.rb [hunk 4]
```diff
@@ -58,15 +58,15 @@
       bob_id = people(:bob).id
       AdvancedTraining.create(description: 'test1', year_from: 2000, month_from: 1, person_id: bob_id)
       AdvancedTraining.create(description: 'test2', year_from: 2016, month_from: 1, year_to: 2030, month_to: 1, person_id: bob_id)
```
Comment (Robin481) [inline spec/models/advanced_training_spec.rb:60]: Cover all cases

**Review by Robin481 — CHANGES_REQUESTED** [review]
:rock:

File: app/helpers/date_helper.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module DateHelper
+  def date_range_label(date_range)
```
Comment (Robin481) [inline app/helpers/date_helper.rb:4]: You are not passing a `date_range` you are passing a model should therefore probably be named differently. Maybe `model_with_dates` or something

**Review by Robin481 — CHANGES_REQUESTED** [review]
as discussed

File: app/controllers/person_relations_controller.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
```
Comment (mtnstar) [inline app/controllers/person_relations_controller.rb:1]: People::RelationsshipsController ?
People::RelationsBaseController ?

PersonRelationsBaseController ?

maybe the name Base would indicate better that this is an abstract controller ?!

put it in People namespace?

just some ideas/inputs ... we might also discuss it

File: app/controllers/person_relations_controller.rb [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    super(:location => person_path(person)) do |format, success|
+      format.turbo_stream { render turbo_stream: turbo_stream.remove(entry) } if success
+    end
+  end
+
+  private
+
+  def entry
+    relation = super
+    relation.person = person unless person == relation.person
```
Comment (mtnstar) [inline app/controllers/person_relations_controller.rb:32]: shouldn't it just raise an exception if person.id is relation.person_id?

shouldn't this be correctly assigned by assign_attributes (dry crud)

File: app/controllers/person_relations_controller.rb [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    relation
+  end
+
+  def person
```
Comment (mtnstar) [inline app/controllers/person_relations_controller.rb:36]: do we need to fetch the complete person here? can't we just use the person's id ?

File: app/helpers/select_helper.rb [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module SelectHelper
+  def select_when_availabale(obj)
+    selected = obj ? obj.id : ''
+    prompt = obj ? false : true
+    { :selected => selected, :prompt => prompt, :disabled => '' }
```
Comment (mtnstar) [inline app/helpers/select_helper.rb:7]:
```suggestion
    { selected: selected, prompt: obj.nil?, disabled: '' }
```

**Review by mtnstar — CHANGES_REQUESTED** [review]
looks pretty solid :rocket: :partying_face: 

please make sure you're not using legacy hash syntax ...

{ a: 'b' } instead of { :a => 'b' }

some other things to check

File: app/controllers/crud_controller.rb [hunk 10]
```diff
@@ -182,8 +182,8 @@ def render_on_success(format, **options)
   end
 
   def render_on_unsaved(format, **options)
-    format.turbo_stream { render options[:render_on_unsaved], status: options[:status] }
     format.html { render_or_redirect_on_unsaved(**options) }
+    format.turbo_stream { render options[:render_on_unsaved], status: options[:status] }
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply app/controllers/crud_controller.rb:186]: yes, with the default order it isn't possible to render turbostreams.haml

File: app/controllers/person_relations_controller.rb (hunk 6 above)
Comment (kcinay055679) [reply app/controllers/person_relations_controller.rb:1]: i took the name from the old controller "Api::PersonRelationsController"

File: app/controllers/person_relations_controller.rb [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    redirect_to person_path(entry.person)
+  end
+
+  def show
+    redirect_to person_path(entry.person)
+  end
+
+  def create
+    super(:location => person_path(person)) do |format, success|
+      format.turbo_stream { render 'save_and_new' } if success && params.key?(:'save-and-new')
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply app/controllers/person_relations_controller.rb:19]: i don't think i can use the turbo:before-render callback in this case

**Review by Robin481 — APPROVED** [review]
👁👄👁
Amazing

**Pull request merged**